content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1460 @@
+            </div>
+            <div>
+                <button type="submit">Confirm</button>
+            </div>
+        </form>
+    </dialog>
+
+    <dialog id="password-dialog">
+        <form method="dialog">
+            <div>
+                <label for="password-input">Enter password (min-length 5):</label>
+                <input type="text" id="password-input" name="password-input" />
+            </div>
+            <div>
+                <button type="button">Change Seed</button>
+                <button type="submit">Submit</button>
+            </div>
+        </form>
+    </dialog>
+```
+
+6. In `view/renderer.js`, replace the `openAccountAddressDialog` part at the top:
+
+```javascript
+// Remove the following section in Step 5
+document.addEventListener('DOMContentLoaded', openAccountAddressDialog);
+
+function openAccountAddressDialog(){
+    const accountAddressDialog = document.getElementById('account-address-dialog');
+    const accountAddressInput = accountAddressDialog.querySelector('input');
+    const submitButton = accountAddressDialog.querySelector('button[type="submit"]');
+
+    submitButton.addEventListener('click', () => {
+        const address = accountAddressInput.value;
+        window.electronAPI.onEnterAccountAddress(address)
+        accountAddressDialog.close()
+    });
+
+    accountAddressDialog.showModal()
+}
+```
+
+With this code:
+
+```javascript
+window.electronAPI.onOpenSeedDialog((_event) => {
+    const seedDialog = document.getElementById('seed-dialog');
+    const seedInput = seedDialog.querySelector('input');
+    const submitButton = seedDialog.querySelector('button[type="submit"]');
+
+    submitButton.addEventListener('click', () => {
+        const seed = seedInput.value;
+        window.electronAPI.onEnterSeed(seed)
+        seedDialog.close()
+    });
+
+    seedDialog.showModal()
+})
+
+window.electronAPI.onOpenPasswordDialog((_event) => {
+    const passwordDialog = document.getElementById('password-dialog');
+    const passwordInput = passwordDialog.querySelector('input');
+    const submitButton = passwordDialog.querySelector('button[type="submit"]');
+    const changeSeedButton = passwordDialog.querySelector('button[type="button"]')
+
+    submitButton.addEventListener('click', () => {
+        const password = passwordInput.value;
+        window.electronAPI.onEnterPassword(password)
+        passwordDialog.close()
+    });
+
+    changeSeedButton.addEventListener('click', () => {
+      passwordDialog.close()
+      window.electronAPI.requestSeedChange()
+    });
+
+    passwordDialog.showModal()
+});
+```
+
+Start up the application:
+
+```console
+npm run start
+```
+
+On first run, It should first prompt you for an account seed and then for a password. After you have created a wallet this way, you should close the application and start it up a second time: On second run it should prompt you for the password, and you should see the same result as in the last step.
+
```

Comment:
I found a small bug when you change the password - for that instance the transactions on display are displayed twice. I'm not sure why: 
<img width="1024" alt="image" src="https://github.com/XRPLF/xrpl-dev-portal/assets/19694186/a163fe0b-b089-4e43-ac6b-2bf329ef1a69">

Reply:
Ok that was a tough one, the same event handler was added twice. I had to restructure the renderer so that the event handler is a named function at the root level and add {once: true} to the options parameter on addEventListener. Learning something every day...

Reply:
Renderer and tutorial (line 966+) have changed, I'll leave it open for now

Reply:
Hahaha, small bugs from deep causes :P